asset.py

```diff
@@ -0,0 +1,41 @@
+import mimetypes
+import api
+import util
+
+
+def initialize_binary_asset(path, **kwargs):
+    with open(util.walk_up_find(path, 'rb')) as image_stream:
```

Comment:
why limit this to images?  Couldn't this also create json assets?

Reply:
This function is specifically for reading binary files.

We could add another convenience method for reading text files, but that doesn't seem as useful.  `initialize_asset` can be called directly, or the user can call `api.create_asset`.